algopy_testing/src/algopy_testing/models/account.py

```diff
@@ -0,0 +1,157 @@
+from __future__ import annotations
+
+from dataclasses import dataclass
+from typing import TYPE_CHECKING, Self, TypedDict, TypeVar
+
+import algosdk
+
+from algopy_testing.primitives.bytes import Bytes
+from algopy_testing.utils import as_string
+
+if TYPE_CHECKING:
+    import algopy
+
+
+T = TypeVar("T")
+
+
+class AccountFields(TypedDict, total=False):
+    balance: algopy.UInt64
+    min_balance: algopy.UInt64
+    auth_address: algopy.Account
+    total_num_uint: algopy.UInt64
+    total_num_byte_slice: algopy.Bytes
+    total_extra_app_pages: algopy.UInt64
+    total_apps_created: algopy.UInt64
+    total_apps_opted_in: algopy.UInt64
+    total_assets_created: algopy.UInt64
+    total_assets: algopy.UInt64
+    total_boxes: algopy.UInt64
+    total_box_bytes: algopy.UInt64
+
+
+@dataclass()
+class Account:
+    _public_key: str
+
+    def __init__(self, value: str | Bytes = algosdk.constants.ZERO_ADDRESS, /):
+        if not isinstance(value, (str | Bytes)):
+            raise TypeError("Invalid value for Account")
+        if isinstance(value, Bytes):
```

Comment:
I think the underlying value stored here should be the bytes, and convert to the human readable form as required. However we should check the bytes is of length 32 to match puya.